src/raft/raft.rs

```diff
@@ -600,16 +599,23 @@ impl<T: Storage> Raft<T> {
     // the commit index changed (in which case the caller should call
     // r.bcast_append).
     pub fn maybe_commit(&mut self) -> bool {
-        // TODO: optimize
-        let mut mis = Vec::with_capacity(self.prs.len());
-        for p in self.prs.values() {
-            mis.push(p.matched);
-        }
-        // reverse sort
-        mis.sort_by(|a, b| b.cmp(a));
-        let mci = mis[self.quorum() - 1];
-        let term = self.term;
-        self.raft_log.maybe_commit(mci, term)
+        let mci = if self.prs.len() <= 5 {
```

Comment:
why < 5 here?

Reply:
So the matched IDs can be put in a [0; 5] array. Generally we only have 3 or 5 replicas.

Reply:
How about `smallvec`[1] ?

[1] https://github.com/servo/rust-smallvec

Reply:
Why use it? I think the final code should look almost the same.

Reply:
I think we should add a benchmark case here.

Reply:
The code will hardly change, an additional benchmark case seems redundant. My benchmark shows that use array is about 50% quicker.

When len is 3:
test bench_commit_array ... bench:          13 ns/iter (+/- 0)
test bench_commit_vec   ... bench:          28 ns/iter (+/- 0)

When len is 5:
test bench_commit_array ... bench:          20 ns/iter (+/- 0)
test bench_commit_vec   ... bench:          35 ns/iter (+/- 0)